_posts/2016-04-25-watson-and-rails.md

```diff
@@ -0,0 +1,213 @@
+    # in this case the questions from the Assignment.
+    @data.values[3].map { |item| [item["title"], item["content"][0]["text"]] }
+  end
+
+  private
+
+  def get_doc
+
+    # Connecting to the API with authorizations
+
+    base_url = "https://gateway.watsonplatform.net"
+    version = "2015-12-15"
+    path = "/document-conversion/api/v1/convert_document?version=#{version}"
+    basic_auth_username = ENV["WATSON_USERNAME"]
+    basic_auth_password = ENV["WATSON_PASSWORD"]
+    # Env variables in your .env file, given to you by signing up with Watson
+
+    conn = Faraday.new(url: base_url) do |faraday|
+      faraday.request :multipart #multipart indicates that a file is being uploaded
+      faraday.request :url_encoded
+      faraday.adapter :net_http
+    end
+
+    conn.basic_auth(basic_auth_username, basic_auth_password)
+
+    # Setting the payload configuration and adding your file to the request
+
+    payload = {}
+    if Rails.env.production?
+      payload = {
+        config: Faraday::UploadIO.new("#{::Rails.root}/lib/WatsonApi/config.json", "application/json"),
+          # config tells Watson what you want your document converted to (JSON/HTML/PLAIN TEXT)
+          # This is in a separate file config.json (see below)
+        file: Faraday::UploadIO.new(open(@doc), "application/pdf")
+          # In production, calling open on the @doc url with open-uri allows us
+          # to open the url as if it were a file
+      }
+    else
+      payload = {
+        config: Faraday::UploadIO.new("#{::Rails.root}/lib/WatsonApi/config.json", "application/json"),
+        file: Faraday::UploadIO.new(@doc, "application/pdf")
+      	# This is the file and type to be uploaded. In development, @doc is a
+      	# local file path so it can be sent without "opening" it first
+      }
+    end
+
+    response = conn.post(path, payload)
+    json_response_body = JSON.parse(response.body)
+  end
+end
+```
+
+The document conversion API
+[docs](http://www.ibm.com/smarterplanet/us/en/ibmwatson/developercloud/document-conversion/api/v1/#convert-document)
+have a lot more information about configuring your request. For configuring the
+file type you want Watson to return (in this case, JSON), the
+`conversion_target` can be set to "answer_units", "normalized_html", or
+"normalized_text".
+
+```ruby
+lib/WatsonApi/config.json
+
+{ "conversion_target" : "ANSWER_UNITS" }
+```
+
+
+Because I was using an `Assignment` model to display assigned work to students,
+my method to parse the returned data was called `questions`. I use this method,
+seen above in the `WatsonApi` wrapper, on the assignment's show page to display
+the converted document:
+
+```html
+app/views/assignments/show.html.erb
+
+<% @document.questions.each do |item| %>
+  <li>
+    <h4><%= item[0] %></h4>
+    <p><%= item[1] %></p>
+  </li>
+<% end %>
+```
+Where `item[0]` was a header or question and `item[1]` was the text beneath it.
```

Comment:
Same problem as above, if I had the JSON response to show and the headers/body they refer to, pictures would be a nice addition here